image_proc/src/nodelets/resize.cpp

```diff
@@ -31,27 +31,35 @@
 *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 *  POSSIBILITY OF SUCH DAMAGE.
 *********************************************************************/
+#include <boost/make_shared.hpp>
+#include <boost/version.hpp>
+#if ((BOOST_VERSION / 100) % 1000) >= 53
+#include <boost/thread/lock_guard.hpp>
+#endif
+
 #include <cv_bridge/cv_bridge.h>
 #include <dynamic_reconfigure/server.h>
 #include <nodelet/nodelet.h>
-#include <nodelet_topic_tools/nodelet_lazy.h>
 #include <ros/ros.h>
 #include <sensor_msgs/CameraInfo.h>
 #include <sensor_msgs/Image.h>
+#include <image_transport/image_transport.h>
 
 #include "image_proc/ResizeConfig.h"
 
 namespace image_proc
 {
 
-class ResizeNodelet : public nodelet_topic_tools::NodeletLazy
+class ResizeNodelet : public nodelet::Nodelet
```

Comment:
why do you need to change from `NodeletLazy` ?

Reply:
I based my work on #307 where doisyg changed it to `Nodelet`.
I support his decision for consistency with the other nodelets from image_proc.
Additionally the `NodeletLazy` class is not documented that well so I prefer choosing the commonly used `Nodelet` class.
Do you have any idea what could be the benefit of the `NodeletLazy` class?

Reply:
@knorth55 Having looked into his code I see why it was chosen: It automatically unsubscribes topics if no one is subscribing the output of the node. The problem is that this class has a protected method `advertise` which must be used to advertise topics. This prohibits of a `image_transport::Publisher` which is the main goal of this PR.